Bug with Mixin usage

Check [this repo](https://github.com/boris-petrov/glint-issues) (actually [this](https://github.com/boris-petrov/glint-issues/commit/556fe399e1014e73746a24851a8095df50a56651) specific commit). It adds a mixin and a component that uses it.

Running `./node_modules/.bin/tsc --noEmit` leads to no errors. However, running `./node_modules/.bin/glint` causes:

```
app/components/test-component.ts:4:44 - error TS4020: 'extends' clause of exported class 'TestComponent' has or is using private name 'ICalendar'.

4 export default class TestComponent extends Component.extend(Calendar) {
                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~

app/components/test-component.ts:4:44 - error TS4020: 'extends' clause of exported class 'TestComponent' has or is using private name 'ICalendar'.

4 export default class TestComponent extends Component.extend(Calendar) {
                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~
```

There shouldn't be a difference here as there are is no HBS involved so `Glint` should behave exactly the same as `tsc`.

P.S. Exporting the `ICalendar` interface fixes the errors however I still believe this is a bug because of the discrepancy between the two type-checkers.

--- a/packages/core/src/cli/perform-check.ts
+++ b/packages/core/src/cli/perform-check.ts
@@ -20,7 +20,7 @@ export function performCheck(
 
   program.emit();
 
-  let diagnostics = collectDiagnostics(program, transformManager);
+  let diagnostics = collectDiagnostics(program, transformManager, parsedConfig.options);
   for (let diagnostic of diagnostics) {
     console.error(transformManager.formatDiagnostic(diagnostic));
   }
@@ -30,13 +30,14 @@ export function performCheck(
 
 function collectDiagnostics(
   program: ts.Program,
-  transformManager: TransformManager
+  transformManager: TransformManager,
+  options: ts.CompilerOptions
 ): Array<ts.Diagnostic> {
   return [
     ...program.getSyntacticDiagnostics(),
     ...transformManager.getTransformDiagnostics(),
     ...program.getSemanticDiagnostics(),
-    ...program.getDeclarationDiagnostics(),
+    ...(options.declaration ? program.getDeclarationDiagnostics() : []),
   ];
 }
 

